src/Component.php

```diff
@@ -2,15 +2,53 @@
 
 declare(strict_types=1);
 
-namespace MyComponent;
+namespace Keboola\StorageWriter;
 
 use Keboola\Component\BaseComponent;
+use Keboola\Component\UserException;
+use Keboola\Csv\CsvFile;
+use Keboola\StorageApi\Client;
+use Keboola\StorageApi\ClientException;
 
 class Component extends BaseComponent
 {
     public function run(): void
     {
-        // @TODO implement
+        /** @var Config $config */
+        $config = $this->getConfig();
+        $client = new Client(['token' => $config->getToken(), 'url' => $config->getUrl()]);
+        $tokenInfo = $client->verifyToken();
+        if (!$tokenInfo['bucketPermissions']) {
+            throw new UserException("The token has too broad permissions.");
+        }
+        $bucket = $config->getBucket();
+        if (empty($tokenInfo['bucketPermissions'][$bucket]) || ($tokenInfo['bucketPermissions'][$bucket] != 'write')) {
```

Comment:
Zároveň zkontrolovat že to je jediný bucket do kterého může zapisovat? A pokud tomu tak není skončit chybou?